app/services/foreman_discovery/ui_notifications/new_host.rb

```diff
@@ -7,7 +7,13 @@ class NewHost < ::UINotifications::Base
       private
 
       def create
-        add_notification if update_notifications.zero?
+        if update_notifications.zero?
+          add_notification
+        else
+          Notification.where(notification_blueprint: blueprint).each do |notification|
+            notification.notification_recipients.update_all(seen: false)
```

Comment:
What about the expiry? They could be expired already.

IMHO you should update them both (seen and expiry), and then maybe the right fix is in core

Reply:
update_notifications already does the mass_update_expiry. I don't know if we should always also mark as unseen when updating the expiry?

Reply:
Then why loop twice? Imho just change it at the same query / scope.

Reply:
I don't mind moving this into `mass_update_expiry`, but do you think we should always mark all as unread when updating expiry?

Reply:
Can we chain update_all? Eg is it a scope?

Reply:
what do you mean by chain it? it return the number of rows modified, and we can't do something like `blueprint.notifications.notification_recipients`. We could add `has_many_through` on the blueprint to get all of the recipients but we'd still need to mark them as unread.

Reply:
How about we add a "mark as unread" call to core?

Reply:
I think the notification stack should have `add_or_update_notification` call. Anyway, there is a random failure, please rebase on develop.

Reply:
Opened https://github.com/theforeman/foreman/pull/4970 to fix this in core, once that is merged i will update this pr accordingly.